File: app/src/main/java/com/drdisagree/iconify/ui/views/LoadingDialog.java
package com.drdisagree.iconify.ui.views;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.drdisagree.iconify.R;

public class LoadingDialog extends AppCompatActivity {

    Context context;
    Dialog dialog;

    public LoadingDialog(Context context) {
        this.context = context;
    }

    public void show(String title) {
        if (dialog != null)
            dialog.dismiss();

        dialog = new Dialog(context);
        dialog.setContentView(R.layout.view_loading_dialog);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.setCancelable(false);
        dialog.setOnCancelListener(null);
        dialog.setCanceledOnTouchOutside(false);

        TextView text = dialog.findViewById(R.id.title);
        text.setText(title);

        dialog.create();
        dialog.show();
    }

    public void show(String title, boolean cancellable) {
        if (dialog != null)
            dialog.dismiss();

        dialog = new Dialog(context);
        dialog.setContentView(R.layout.view_loading_dialog);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.setCancelable(cancellable);
        dialog.setOnCancelListener(null);
        dialog.setCanceledOnTouchOutside(cancellable);

        TextView text = dialog.findViewById(R.id.title);
        text.setText(title);

        dialog.create();
        dialog.show();
    }

    public void hide() {
        if ((dialog != null) && dialog.isShowing())
            dialog.dismiss();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (dialog != null && !isFinishing()) {
            dialog.dismiss();
        }
    }
}

File: app/src/main/java/com/drdisagree/iconify/ui/dialogs/LoadingDialog.kt
package com.drdisagree.iconify.ui.dialogs

import android.app.Dialog
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.widget.TextView
import com.drdisagree.iconify.R
import com.drdisagree.iconify.ui.base.BaseActivity

class LoadingDialog(var context: Context) : BaseActivity() {
    var dialog: Dialog? = null

    fun show(title: String?) {
        if (dialog != null) dialog!!.dismiss()

        dialog = Dialog(context)
        dialog!!.setContentView(R.layout.view_loading_dialog)
        dialog!!.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        dialog!!.setCancelable(false)
        dialog!!.setOnCancelListener(null)
        dialog!!.setCanceledOnTouchOutside(false)

        val text = dialog!!.findViewById<TextView>(R.id.title)
        text.text = title

        dialog!!.create()
        dialog!!.show()
    }

    fun show(title: String?, cancellable: Boolean) {
        if (dialog != null) dialog!!.dismiss()

        dialog = Dialog(context)
        dialog!!.setContentView(R.layout.view_loading_dialog)
        dialog!!.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        dialog!!.setCancelable(cancellable)
        dialog!!.setOnCancelListener(null)
        dialog!!.setCanceledOnTouchOutside(cancellable)

        val text = dialog!!.findViewById<TextView>(R.id.title)
        text.text = title

        dialog!!.create()
        dialog!!.show()
    }

    fun dismiss() {
        dialog?.dismiss()
    }

    fun hide() {
        if (dialog?.isShowing == true) {
            dialog?.dismiss()
        }
    }

    public override fun onDestroy() {
        dismiss()
        super.onDestroy()
    }
}
